src/pages/CartRouter/Cart/Cart.js

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import CartProduct from './CartProduct/CartProduct';
+import './Cart.scss';
+
+const Cart = () => {
+  const [cartList, setCartList] = useState([]);
+  const [checkList, setCheckList] = useState([]);
+  const [total, setTotal] = useState(0);
+
+  const sum = data => {
+    let totalList = data.result.map(
+      product => product.quantity * product.price
+    );
+
+    let sum = 0;
+    for (let i of totalList) {
+      sum += i;
+    }
+    setTotal(sum);
+  };
```

Comment:
* sum 함수에서는 product의 주문제품당 가격을 배열로 반환하는 기능과
* 반환된 배열을 더해서 setTotal 해주는 기능 하나 이렇게 두가지가 같이 동작하고 있네요
* 하나의 함수에서는 한가지 기능만 할 수 있도록 관심사의 분리를 하는게 유지보수와 모듈화 하기에 좋습니다.
```suggestion
  const sumOrderdPrice = data => {
    let totalList = data.result.map(
      product => product.quantity * product.price
    );
}
const getTotalPrice = ()=>{
    let sum = 0;
    for (let i of totalList) {
      sum += i;
      
}

    
    setTotal(sum);
  
```